prov/efa/src/efa_fabric.c

```diff
@@ -471,6 +473,135 @@ static int efa_alloc_fid_nic(struct fi_info *fi, struct efa_context *ctx,
 	return ret;
 }
 
+#else // _WIN32
+
+static int efa_alloc_fid_nic(struct fi_info *fi, struct efa_context *ctx,
+			     struct efa_device_attr *efa_device_attr,
+			     struct ibv_port_attr *port_attr)
+{
+	struct fi_device_attr *device_attr;
+	struct fi_link_attr *link_attr;
+	struct fi_bus_attr *bus_attr;
+	void *src_addr;
+	int name_len;
+	int ret;
+
+	/* Sets nic ops and allocates basic structure */
+	fi->nic = ofi_nic_dup(NULL);
+	if (!fi->nic)
+		return -FI_ENOMEM;
+
+	device_attr = fi->nic->device_attr;
+	bus_attr = fi->nic->bus_attr;
+	link_attr = fi->nic->link_attr;
+
+	/* fi_device_attr */
+	device_attr->name = strdup(ctx->ibv_ctx->device->name);
+	if (!device_attr->name) {
+		ret = -FI_ENOMEM;
+		goto err_free_nic;
+	}
+
+	ret = asprintf(&device_attr->device_id, "0x%x",
+		       efa_device_attr->ibv_attr.vendor_part_id);
+	/* ofi_nic_close will free all attributes of the fi_nic struct */
+	if (ret < 0) {
+		ret = -FI_ENOMEM;
+		goto err_free_nic;
+	}
+
+	/*
+	 * On Windows, there is no sysfs. We use hw_ver field of ibv_attr to obtain it
+	 */
+	ret = asprintf(&device_attr->device_version, "%u", efa_device_attr->ibv_attr.hw_ver);
```

Comment:
The difference between the original function and this function is on `driver_version` and `driver`, right? can you wrap them in a function like `xxx_setup_driver` which has two implementations, and have `efa_fid_alloc_nic` call it?

Reply:
Yes, please.

Reply:
will do